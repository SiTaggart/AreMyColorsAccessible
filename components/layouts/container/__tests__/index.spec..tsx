/* eslint-env jest */
/// <reference types="jest" />

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Container } from '..';

describe('Container', (): void => {
  it('renders without crashing', (): void => {
    ReactDOM.render(
      <Container>
        <div>children</div>
      </Container>,
      document.createElement('div')
    );
    const containerCmp = renderer
      .create(
        <Container>
          <div>children</div>
        </Container>
      )
      .toJSON();
    expect(containerCmp).toMatchSnapshot();
  });

  it('should set home class with home variant', (): void => {
    const containerCmp = renderer
      .create(
        <Container variant="home">
          <div>children</div>
        </Container>
      )
      .toJSON();
    expect(containerCmp).toMatchSnapshot();
  });

  it('should set about class with about variant', (): void => {
    const containerCmp = renderer
      .create(
        <Container variant="about">
          <div>children</div>
        </Container>
      )
      .toJSON();
    expect(containerCmp).toMatchSnapshot();
  });

  it('should set palette class with palette variant', (): void => {
    const containerCmp = renderer
      .create(
        <Container variant="palette">
          <div>children</div>
        </Container>
      )
      .toJSON();
    expect(containerCmp).toMatchSnapshot();
  });
});
