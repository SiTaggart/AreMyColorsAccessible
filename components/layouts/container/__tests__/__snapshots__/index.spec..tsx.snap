// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders without crashing 1`] = `
.emotion-0 {
  color: ;
  padding-top: ;
  position: relative;
}

@media (min-width: 768px) {
  .emotion-0 {
    min-height: ;
  }
}

<main
  className="emotion-0 emotion-1"
>
  <div>
    children
  </div>
</main>
`;

exports[`Container should set about class with about variant 1`] = `
.emotion-0 {
  color: ;
  padding-top: 3rem;
  position: relative;
}

@media (min-width: 768px) {
  .emotion-0 {
    min-height: ;
  }
}

<main
  className="emotion-0 emotion-1"
>
  <div>
    children
  </div>
</main>
`;

exports[`Container should set home class with home variant 1`] = `
.emotion-0 {
  color: ;
  padding-top: ;
  position: relative;
}

@media (min-width: 768px) {
  .emotion-0 {
    min-height: 85%;
  }
}

<main
  className="emotion-0 emotion-1"
>
  <div>
    children
  </div>
</main>
`;

exports[`Container should set palette class with palette variant 1`] = `
.emotion-0 {
  color: #343334;
  padding-top: ;
  position: relative;
}

@media (min-width: 768px) {
  .emotion-0 {
    min-height: ;
  }
}

<main
  className="emotion-0 emotion-1"
>
  <div>
    children
  </div>
</main>
`;
